### PR TITLE
Adding start of analyzer

### PR DESCRIPTION
**Pull request by stimms**
This PR contains a (I think) good start on building out an analyzer to help people with their durable functions. The code is taken from https://github.com/stimms/DurableFunctionsAnalyzer and some more details are available in this blog post https://blog.simontimms.com/2019/02/17/2019-02-17-durable_functions_analyzer/

I imagine some significant work will be needed to bring it up to the quality guidelines for this project and I'm happy to help out with some of that. This should be considered a work in progress.

### CODE REVIEW

File: src/Analyzer/DurableFunctionsAnalyzer/DurableFunctionsAnalyzer/AnalyzerRegistration.cs.orig [hunk 1]
```diff
@@ -0,0 +1,60 @@
+using DurableFunctionsAnalyzer.Analyzers;
```
Comment (cgillum) [inline src/Analyzer/DurableFunctionsAnalyzer/DurableFunctionsAnalyzer/AnalyzerRegistration.cs.orig]: Can this file be removed? It looks like an artifact from merge conflict resolution.

File: src/Analyzer/DurableFunctionsAnalyzer/DurableFunctionsAnalyzer/DurableFunctionsAnalyzer.csproj [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    <GeneratePackageOnBuild>True</GeneratePackageOnBuild>
+  </PropertyGroup>
+
+  <PropertyGroup>
+    <PackageId>DurableFunctionsAnalyzer</PackageId>
+    <PackageVersion>1.0.0.0</PackageVersion>
+    <Authors>stimms</Authors>
+    <PackageLicenseUrl>https://github.com/stimms/DurableFunctionsAnalyzer/blob/master/LICENSE</PackageLicenseUrl>
+    <PackageProjectUrl>https://github.com/stimms/DurableFunctionsAnalyzer</PackageProjectUrl>
+    <RepositoryUrl>https://github.com/stimms/DurableFunctionsAnalyzer</RepositoryUrl>
```
Comment (cgillum) [inline src/Analyzer/DurableFunctionsAnalyzer/DurableFunctionsAnalyzer/DurableFunctionsAnalyzer.csproj]: These URLs will need to be updated.

File: src/Analyzer/DurableFunctionsAnalyzer/DurableFunctionsAnalyzer/DurableFunctionsAnalyzer.csproj [hunk 3]
```diff
@@ -0,0 +1,39 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>netstandard1.3</TargetFramework>
```
Comment (cgillum) [inline src/Analyzer/DurableFunctionsAnalyzer/DurableFunctionsAnalyzer/DurableFunctionsAnalyzer.csproj]: We've been targeting `netstandard2.0` for our projects, so I suggest we do the same here for consistency.

File: src/Analyzer/DurableFunctionsAnalyzer/DurableFunctionsAnalyzer.Test/NameAnalyzerTests.cs [hunk 4]
```diff
@@ -0,0 +1,348 @@
...
+            //using System.Threading.Tasks;
+            //using System.Diagnostics;
+
+            //namespace ConsoleApplication1
+            //{
+            //    class TYPENAME
+            //    {   
+            //    }
+            //}";
+            //        VerifyCSharpFix(test, fixtest);
```
Comment (cgillum) [inline src/Analyzer/DurableFunctionsAnalyzer/DurableFunctionsAnalyzer.Test/NameAnalyzerTests.cs]: Can this commented-out code be removed? I believe there were a few other places with commented-out code like this.

**Comment by ConnorMcMahon** [comment]
@stimms, 

@amdeel has been working on adding lots of analyzers, and has used some of the work you have started here in her work. Unfortunately, we are targeting this work for the beta2 release of Durable Functions V2, meaning it will eventually be merged into our `v2` branch.

Would you mind retargeting this PR against our `v2` branch. That way Amanda can add commits on top of yours, and you will still get credit in the commit history when we merge this.

**Comment by cgillum** [comment]
Thanks again @stimms for this contribution! We ended up copying your code into a different PR, which we just merged (https://github.com/Azure/azure-functions-durable-extension/pull/949). It includes your work plus a bunch of new analyzers created by @amdeel. We added you as a co-contributor using GitHub's `Co-authored-by` feature so that you're given credit for the work. I'll go ahead and close out this PR.

**Pull request closed without merging**